pilopt/src/lib.rs

```diff
@@ -397,54 +399,64 @@ fn remove_trivial_selectors<T: FieldElement>(pil_file: &mut Analyzed<T>) {
 fn extract_constant_lookups<T: FieldElement>(pil_file: &mut Analyzed<T>) {
     let mut new_identities = vec![];
     for identity in &mut pil_file.identities.iter_mut() {
-        if let Identity::Lookup(LookupIdentity {
-            source,
-            left,
-            right,
-            ..
-        }) = identity
-        {
-            let mut extracted = HashSet::new();
-            for (i, (l, r)) in left
-                .expressions
-                .iter()
-                .zip(&right.expressions)
-                .enumerate()
-                .filter_map(|(i, (l, r))| {
-                    if let AlgebraicExpression::Number(n) = r {
-                        Some((i, (l, n)))
-                    } else {
-                        None
-                    }
-                })
-            {
-                // TODO remove clones
-                let l_sel = left
-                    .selector
-                    .clone()
-                    .unwrap_or_else(|| AlgebraicExpression::from(T::one()));
-                let r_sel = right
-                    .selector
-                    .clone()
-                    .unwrap_or_else(|| AlgebraicExpression::from(T::one()));
-                let pol_id = (l_sel * l.clone()) - (r_sel * AlgebraicExpression::from(*r));
-                new_identities.push((simplify_expression(pol_id), source.clone()));
-
-                extracted.insert(i);
+        match identity {
+            Identity::Lookup(LookupIdentity {
+                source,
+                left,
+                right,
+                ..
+            })
+            | Identity::PhantomLookup(PhantomLookupIdentity {
```

Comment:
Here I'm less sure: Should `pilopt` treat Phantom lookups like lookups?

Reply:
Yes, pilopt shouldn't care in any way.